### PR TITLE
Promote fix

### PR DESCRIPTION
**Pull request by eidelmanjonathan**
In the previous version of the scripts, if you promoted a snapshot from one account to the other, it was rebuilding the snapshot and giving it a new ID. This defeats the purpose of having a shared build account and is fixed here

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: ci/snapshot/snapshot-update [hunk 1]
```diff
@@ -230,39 +233,51 @@ def main():
...
-        build_s3_client = session.client("s3")
-        build_ecr_client = session.client("ecr")
         wait_for_pipeline_completion("Build-CBMC-Linux-Pipeline", pipeline_client)
         wait_for_pipeline_completion("Build-Docker-Pipeline", pipeline_client)
         wait_for_pipeline_completion("Build-Viewer-Pipeline", pipeline_client)
         wait_for_pipeline_completion("Build-Batch-Pipeline", pipeline_client)
         package_filenames = get_package_filenames_from_s3(build_s3_client, build_ecr_client)
         update_and_write_snapshot(package_filenames, snapshot)
 
-    # Otherwise we should propose the most recent snapshot
```
Comment (mwwhalen) [inline ci/snapshot/snapshot-update]: What if both flags init_build_account and init_proof_account are set?

Comment (eidelmanjonathan) [reply ci/snapshot/snapshot-update]: This is not the correct usage, I will have it throw an error

Comment (eidelmanjonathan) [reply ci/snapshot/snapshot-update]: The intention is that we would first run snapshot-update with --init-build-account flag, which would build all the globals in the shared tools account, create a new snapshot in the build account, and deploy that snapshot in the proof account.

Next, if we needed to initialize a new proof account to share the build account we would use the --init-proof-account flag which would create a new snapshot in the build account and deploy it in the proof account.

If we want to propose and deploy and updated snapshot for an existing proof account (that has github.yaml deployed), can run snapshot-update without extra flags.

Finally, to promote a snapshot from one account to another, we use --promote-from which takes the source account as an argument. This does not build a new snapshot

File: ci/snapshot/snapshot-update [hunk 2]
```diff
@@ -230,39 +233,55 @@ def main():
     args = parse_args()
     profile = args.profile
     build_profile = args.build_profile
+    session = boto3.session.Session(profile_name=build_profile)
+    build_s3_client = session.client("s3")
+    build_ecr_client = session.client("ecr")
+    snapshot = snapshott.Snapshot(filename="snapshot.json")
+
```
Comment (mww-aws) [inline ci/snapshot/snapshot-update]: Thanks for fixing this.

**Pull request merged**